backend/src/test/java/fr/inra/urgi/datadiscovery/dao/DocumentIndexSettings.java

```diff
@@ -1,39 +1,25 @@
 package fr.inra.urgi.datadiscovery.dao;
 
-import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.util.Map;
-
-import org.elasticsearch.common.settings.Settings;
+import co.elastic.clients.elasticsearch.indices.IndexSettings;
 
 /**
  * Utility class to load the settings from the resources
  * @author JB Nizet
  */
 public class DocumentIndexSettings {
-    public static Settings createSettings(String profile) {
-        try {
-            String appName = profile.substring(0, profile.lastIndexOf("-"));
-            return Settings.builder().loadFromStream(
-                    appName + "/settings.json",
-                    DocumentIndexSettings.class.getResourceAsStream(appName + "/settings.json"),
-                    true
-            ).build();
-        }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+    public static IndexSettings createSettings(String profile) {
+        String appName = profile.substring(0, profile.lastIndexOf("-"));
+        return IndexSettings.of(
+            builder ->
+                builder.withJson(DocumentIndexSettings.class.getResourceAsStream(appName + "/settings.json"))
+                       //.providedName(appName + "/settings.json")
```

Comment:
Is this comment needed? here and below

Reply:
Good catch.

The old code passed a "resourceName" argument. I thought the equivalent with the Java client was this `providedName` property. But it turns out that this property is not supported by the version of Elastic that we're currently using for our tests (and that we might have to change, BTW, but I don't know which version of ElasticSearch the INRAE intends to use).
I commented it to make the tests pass and forgot to investigate further.
I've now investigated, and it turns out the old code only used that name to decide if the resource had to be parsed as Yaml or as JSON.
I've removed the comment.